src/managers/KeybindManager.hpp

```diff
@@ -95,6 +95,11 @@ class CKeybindManager {
 
     std::list<SKeybind>                                               m_lKeybinds;
 
+    //since we cant find keycode through keyname in xkb:
+    //on sendshortcut call, we once search for keyname (e.g. "g") the correct keycode (e.g. 42)
+    //and cache it in this map to make sendshortcut calls faster
+    std::unordered_map<std::string, xkb_keycode_t>                    m_mKeyToCodeCache;
```

Comment:
different keyboards can have different layouts, this can break.

Reply:
Done, added the pointer to the current keyboard to the string to differentiate. Didnt use `std::pair<wlr_keyboard*, std:.string>` since hashing doesnt work with this smh